synapse/handlers/auth.py

```diff
@@ -885,6 +889,18 @@ async def _find_user_id_and_pwd_hash(
             )
         return result
 
+    def get_allow_change_password(self) -> bool:
```

Comment:
I don't love this name, but am not sure what's better, maybe only `can_change_password`? 🤷